packages/opentelemetry-node-tracer/test/NodeTracer.test.ts

```diff
@@ -0,0 +1,181 @@
+        binaryFormat: new BinaryTraceContext(),
+        scopeManager: new AsyncHooksScopeManager(),
+      });
+      assert.ok(tracer instanceof NodeTracer);
+    });
+
+    it('should construct an instance with http text format', () => {
+      const tracer = new NodeTracer({
+        httpTextFormat: new HttpTraceContext(),
+        scopeManager: new AsyncHooksScopeManager(),
+      });
+      assert.ok(tracer instanceof NodeTracer);
+    });
+
+    it('should construct an instance with logger', () => {
+      const tracer = new NodeTracer({
+        logger: new NoopLogger(),
+        scopeManager: new AsyncHooksScopeManager(),
+      });
+      assert.ok(tracer instanceof NodeTracer);
+    });
+
+    it('should construct an instance with sampler', () => {
+      const tracer = new NodeTracer({
+        scopeManager: new AsyncHooksScopeManager(),
+        sampler: ALWAYS_SAMPLER,
+      });
+      assert.ok(tracer instanceof NodeTracer);
+    });
+
+    it('should construct an instance with default attributes', () => {
+      const tracer = new NodeTracer({
+        defaultAttributes: {
+          region: 'eu-west',
+          asg: 'my-asg',
+        },
+        scopeManager: new AsyncHooksScopeManager(),
+      });
+      assert.ok(tracer instanceof NodeTracer);
+    });
+  });
+
+  describe('startSpan', () => {
+    it('should start a span with name only', () => {
+      const tracer = new NodeTracer({
+        scopeManager: new AsyncHooksScopeManager(),
+      });
+      const span = tracer.startSpan('my-span');
+      assert.ok(span);
+    });
+
+    it('should start a span with name and options', () => {
+      const tracer = new NodeTracer({
+        scopeManager: new AsyncHooksScopeManager(),
+      });
+      const span = tracer.startSpan('my-span', {});
+      assert.ok(span);
+    });
+
+    it('should return a default span with no sampling', () => {
+      const tracer = new NodeTracer({
+        sampler: NEVER_SAMPLER,
+        scopeManager: new AsyncHooksScopeManager(),
+      });
+      const span = tracer.startSpan('my-span');
+      assert.deepStrictEqual(span, NodeTracer.defaultSpan);
+    });
+
+    // @todo: implement
+    it('should start a Span with always sampling');
+
+    // @todo: implement
+    it('should set default attributes on span');
+  });
+
+  describe('getCurrentSpan', () => {
+    it('should return null with AsyncHooksScopeManager when no span started', () => {
+      const tracer = new NodeTracer({
+        scopeManager: new AsyncHooksScopeManager(),
+      });
+      assert.deepStrictEqual(tracer.getCurrentSpan(), null);
+    });
+  });
+
+  describe('withSpan', () => {
+    it('should run scope with AsyncHooksScopeManager scope manager', done => {
+      const tracer = new NodeTracer({
+        scopeManager: new AsyncHooksScopeManager(),
+      });
+      const span = tracer.startSpan('my-span');
+      tracer.withSpan(span, () => {
+        assert.deepStrictEqual(tracer.getCurrentSpan(), span);
```

Comment:
I think we should call `done()` here

Reply:
I called `done()` down below, if I call here next test(`assert.deepStrictEqual(tracer.getCurrentSpan(), null);`) is not running.

Reply:
couldn't you move the `assert.deepStrictEqual(tracer.getCurrentSpan(), null);` before the with span?
If I understand the test correctly you are checking first that scope by default is null, then you set a span in the scope and assert that you get it back. If this is the case I'd organize the test like:
```js
it('should run scope with AsyncHooksScopeManager scope manager', done => {
  const tracer = new NodeTracer({
    scopeManager: new AsyncHooksScopeManager(),
  });
  assert.deepStrictEqual(tracer.getCurrentSpan(), null);
  const span = tracer.startSpan('my-span');
  tracer.withSpan(span, () => {
    assert.deepStrictEqual(tracer.getCurrentSpan(), span);
    return done();
  });
});

Reply:
I'm just worried that it as `withSpan` is async and id we don't call done there, the assert will run and see the error for sure, but the test suite may be already starting to run another test and the output will be hard to read.

Reply:
`withSpan` looks async, but it's not really. The callback is just a way to constrain the scope activation and automatically catch errors.

Reply:
> If I understand the test correctly you are checking first that scope by default is null, then you set a span in the scope and assert that you get it back.

Right and even I wanted to check `getCurrentSpan()` at the end of the scope. I expected current span to be `null` at the end of the scope.

Reply:
This is not a blocker for now, I would like to merge this PR. Please update this thread if someone have an idea on how to check `getCurrentSpan() === null` at the end of the scope.